packages/neptune-extension/src/common/core/notebook.js

```diff
@@ -0,0 +1,53 @@
+import {
+  getAccessToken,
+  getNotebookData,
+  createNotebook,
+  createCheckpoint,
+} from 'common/api/backend';
+
+export function uploadNotebook(metadata, token, projectId, content) {
+  return new Promise((resolve, reject) => {
+    getAccessToken(token)
+      .then(() => {
+        createNotebook(projectId)
+          .then(({ data: notebookData }) => {
+            createCheckpoint(notebookData.id, content, metadata.path)
+              .then(() => resolve(notebookData))
+              .catch(reject);
+          })
+          .catch(reject);
+      });
+  });
+}
+
+export function uploadCheckpoint(metadata, token, content) {
+  return new Promise((resolve, reject) => {
```

Comment:
It is the same here.
I won't prepare replacement code in this case but based on two cases above you should catch the point.